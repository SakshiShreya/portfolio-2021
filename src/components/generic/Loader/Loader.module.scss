@import "../../../utilities/partials/colorCodes";

$box-border-radius: 15%;
$duration: 1.5s;

.loader {
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
  position: relative;
}

.box {
  position: relative;
  display: block;
  transform-origin: -50% center;
  border-radius: $box-border-radius;

  &::after {
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    right: 0;
    background-color: map-get($map: $orange, $key: "normal");
    border-radius: $box-border-radius;
  }

  &:nth-child(1) {
    &::after {
      animation: color-change $duration ease-in-out infinite alternate;
    }
  }

  @for $i from 1 to 5 {
    &:nth-child(#{$i + 1}) {
      animation: flip-#{$i} $duration ease-in-out infinite alternate;
      &:after {
        background-color: map-get($map: $orange, $key: "light" + $i);
        animation: squidge-#{$i} $duration ease-in-out infinite alternate;
      }
    }
  }
}

@keyframes color-change {
  0% {
    background-color: map-get($map: $orange, $key: "normal");
  }
  100% {
    background-color: map-get($map: $orange, $key: "light5");
  }
}

@for $i from 1 to 5 {
  @keyframes flip-#{$i} {
    0%,
    #{$i * 15%} {
      transform: rotate(0);
    }
    #{$i * 15 + 20%},
    100% {
      transform: rotate(-180deg);
    }
  }

  @keyframes squidge-#{$i} {
    #{$i * 15% - 10%} {
      transform-origin: center bottom;
      transform: scaleX(1) scaleY(1);
    }
    #{$i * 15%} {
      transform-origin: center bottom;
      transform: scaleX(1.3) scaleY(0.7);
    }
    #{$i * 15% + 5%},
    #{$i * 15% + 10%} {
      transform-origin: center bottom;
      transform: scaleX(0.8) scaleY(1.4);
    }
    #{$i * 15% + 25%} {
      transform-origin: center top;
      transform: scaleX(1.3) scaleY(0.7);
    }
    #{$i * 15% + 40%},
    100% {
      transform-origin: center top;
      transform: scaleX(1) scaleY(1);
    }
  }
}

@mixin loader($size) {
  $box-size: $size * 0.12;

  width: $size;

  .box {
    width: $box-size;
    height: $box-size;

    &::after {
      box-shadow: 0px 0px $size/25 0 adjust-color($color: map-get($map: $orange, $key: "normal"), $alpha: 0.4);
    }

    &:nth-child(1) {
      animation: slide-#{$size} $duration ease-in-out infinite alternate;
    }
  }

  @keyframes slide-#{$size} {
    0% {
      transform: translatex(0);
    }
    100% {
      transform: translatex($size - ($box-size * 1.25));
    }
  }
}

.loader_200 {
  @include loader(200px);
}

.loader_60 {
  @include loader(60px);
}
