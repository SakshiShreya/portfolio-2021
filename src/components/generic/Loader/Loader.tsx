import { FunctionComponent } from "react";
import styles from "./Loader.module.scss";

interface LoaderProps {
  size: number;
}

const Loader: FunctionComponent<LoaderProps> = props => {
  console.log(styles);
  return (
    <div className={`${styles.loader} ${styles["loader_" + props.size]}`}>
      <div className={styles.box}></div>
      <div className={styles.box}></div>
      <div className={styles.box}></div>
      <div className={styles.box}></div>
      <div className={styles.box}></div>
    </div>
  );
};

export default Loader;
