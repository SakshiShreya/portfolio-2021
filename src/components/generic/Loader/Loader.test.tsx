import renderer from "react-test-renderer";
import Loader from "./Loader";

describe("test loader for different sizes", () => {
  it("should render loader of size 20 when size={20} is passed in prop", () => {
    const elem = renderer.create(<Loader size={20} />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it("should render loader of size 30 when size={30} is passed in prop", () => {
    const elem = renderer.create(<Loader size={30} />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
