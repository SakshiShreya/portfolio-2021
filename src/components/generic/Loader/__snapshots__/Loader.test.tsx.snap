// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test loader for different sizes should render loader of size 20 when size={20} is passed in prop 1`] = `
<div
  className="loader loader_20"
>
  <div
    className="box"
  />
  <div
    className="box"
  />
  <div
    className="box"
  />
  <div
    className="box"
  />
  <div
    className="box"
  />
</div>
`;

exports[`test loader for different sizes should render loader of size 30 when size={30} is passed in prop 1`] = `
<div
  className="loader loader_30"
>
  <div
    className="box"
  />
  <div
    className="box"
  />
  <div
    className="box"
  />
  <div
    className="box"
  />
  <div
    className="box"
  />
</div>
`;
